Description: Fix column delimiter issues in numsum
 See bug #423667, thanks to Filippo Giunchedi <filippo@debian.org> for the good bug report.
Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=423667
Last-Update: 2007-06-16
Index: num-utils-0.5/numsum
===================================================================
--- numsum	2016-08-06 08:56:16.746570096 +0200
+++ numsum	2016-08-06 08:56:16.742570096 +0200
@@ -28,6 +28,7 @@
 use Getopt::Std;
 #use strict;
 use vars qw/ %opts /;
+use Scalar::Util qw/looks_like_number/;
 
 my ($file, $finalsum, @number_array, @columns_to_print, @column_output, $output_line, $string_seperator, $verbose);
 
@@ -198,6 +199,7 @@
         # try summing 1.99 and -1.90 to understand why
         my $number_a = shift;
         my $number_b = shift;
+	return $number_a if( ! looks_like_number( $number_b ) );
         my $dec_a = ( $number_a =~ /\.([^\.]*)$/ ? length($1) : 0 );
         my $dec_b = ( $number_b =~ /\.([^\.]*)$/ ? length($1) : 0 );
         my $factor = 10 ** ( $dec_a > $dec_b ? $dec_a : $dec_b );
@@ -226,7 +228,7 @@
             #s!(\S+)!$1=~/\D/?0:$1!ge;
           
             # This one works best. 
-            $line =~ s!(\S+)!$1=~/^(\-?[0-9]*\.?[0-9]+)$/?$1:0!ge;
+            #$line =~ s!(\S+)!$1=~/^(\-?[0-9]*\.?[0-9]+)$/?$1:0!ge;
 
             push(@$number_array_ref, $line);
 
@@ -263,11 +265,8 @@
     my $input_array_length = @$input_array_ref;
     my $x;
     for ($x = 0; $x < $input_array_length; $x++) {
-        if (${$sum_array_ref}[$x]) {
+            ${$sum_array_ref}[$x] = 0 if( ! defined( ${$sum_array_ref}[$x] ) );
             ${$sum_array_ref}[$x] = decimals_friendly_sum( ${$sum_array_ref}[$x], ${$input_array_ref}[$x] );
-        } else {
-            ${$sum_array_ref}[$x] = ${$input_array_ref}[$x];
-        }
     }
     return 1;
 }
